File: q1/code.go
package q1

func twoSum(nums []int, target int) []int {
	m := map[int]int{}
	for i, e := range nums {
		if x, found := m[e]; found {
			return []int{x, i}
		}
		m[target - e] = i
	}
	return []int{}
}

File: q1/code.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var m = {}, i = 0, l = nums.length;
    while(i < l) {
        var e = nums[i]|0, x = m[e];
        if (x !== undefined) {
            return [x, i];
        }
        m[target - e] = i|0;
        i=(i+1)|0;
    }
    return [];
};
